### PR TITLE
Securing the obfuscated password file/folder

### PR DESCRIPTION
**Pull request by joergsteinkamp**
Dear all,
Nice and usefull tool. The obfuscated password needs a better protection in my opinion. I inserted two possibilities, how to create either a folder with permission (700) or a file with (600). Nevertheless, I don't know how this would translate inti the Windows world. See also Stackoverflow where I got the code snippet from:
https://stackoverflow.com/questions/36745577/how-do-you-create-in-python-a-file-with-permissions-other-users-can-write
Kind regards,
Jörg

### CODE REVIEW

**Comment by chStaiger** [comment]
Thank you for the code contribution, it is. indeed very useful. We will test whether it will also work in our windows environments. And we will also adjust the iBridges CLI in our API package accordingly.

**Comment by chStaiger** [comment]
@joergsteinkamp  I started our linting and it finds some python errors. E.g. the `os` module is not imported etc. Could you have a look at this please?

**Comment by chStaiger** [comment]
Thanks for the update. Unfortunately the linters are still complaining.
You can check it yourself locally with 

```
ruff ibridgesgui
pylint ibridgesgui
```

**Comment by joergsteinkamp** [comment]
Sorry, I missed inserting the 'import os' in login.py in my github repo. It's now in there. Also the docstring in imperative: 'Creates' and 'Creating' was not accepted.

File: ibridgesgui/login.py [hunk 1]
```diff
@@ -22,6 +23,9 @@
 from ibridgesgui.gui_utils import UI_FILE_DIR
 from ibridgesgui.ui_files.irodsLogin import Ui_irodsLogin
 
+def strictwrite(path, flags, mode='0o600'):
```
Comment (chStaiger) [inline ibridgesgui/login.py:26]:
```suggestion

def strictwrite(path, flags, mode='0o600'):
```

**Review by qubixes — APPROVED** [review]

**Comment by chStaiger** [comment]
Thank you for the update. I tested your new feature on Mac, but I run into an error:

```
ERROR:ibridges-gui:Failed to login: TypeError("'str' object cannot be interpreted as an integer")
Traceback (most recent call last):
  File "/Users/user/test-git/iBridges-GUI/ibridgesgui/login.py", line 113, in login_function
    with open(IRODSA, "w",  encoding="utf-8", opener=strictwrite) as f:
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/user/test-git/iBridges-GUI/ibridgesgui/login.py", line 29, in strictwrite
    return os.open(path, flags, mode)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
TypeError: 'str' object cannot be interpreted as an integer
```

**Comment by chStaiger** [comment]
Thanks for the update. I tried the new feature on Mac but I run into this error:

```
ERROR:ibridges-gui:Failed to login: TypeError("'str' object cannot be interpreted as an integer")
Traceback (most recent call last):
  File "/Users/user/test-git/iBridges-GUI/ibridgesgui/login.py", line 113, in login_function
    with open(IRODSA, "w",  encoding="utf-8", opener=strictwrite) as f:
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/user/test-git/iBridges-GUI/ibridgesgui/login.py", line 29, in strictwrite
    return os.open(path, flags, mode)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
TypeError: 'str' object cannot be interpreted as an integer
```

It might also be nice to have a unit test for this function. Would you like to write that test or would you like us to do it?

File: ibridgesgui/login.py [hunk 2]
```diff
@@ -23,6 +24,10 @@
 from ibridgesgui.ui_files.irodsLogin import Ui_irodsLogin
 
 
+def strictwrite(path, flags, mode='0o600'):
```
Comment (chStaiger) [inline ibridgesgui/login.py:27]:
```suggestion
def strictwrite(path, flags, mode=600):
```

**Review by chStaiger — CHANGES_REQUESTED** [review]
Something seems to be wrong with the mode flag in `os.open`:

```
In [4]: ?os.open
Signature: os.open(path, flags, mode=511, *, dir_fd=None)
Docstring:
Open a file for low level IO.  Returns a file descriptor (integer).
```

**Comment by joergsteinkamp** [comment]
Ok, on Linux I had trouble with integers looking like permissions, the mode was more readable for me in the octal version. 511 is 0o777 in octal, which is the default and would give the default as access rights. 0o600 would be 384 and 0o700 448, respectiively. Use print(0o700) to convert it to integers and oct(511) vice versa.

File: ibridgesgui/login.py (hunk 2 above)
Comment (chStaiger) [inline ibridgesgui/login.py:27]:
```suggestion
def strictwrite(path, flags, mode=0o600):
```

**Review by chStaiger — APPROVED** [review]
Thanks for this very useful and nice contribution!

**Pull request merged**